include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -260,15 +277,33 @@ namespace LinearAlgebra
 #  endif
       }
       if (operation == VectorOperation::insert)
-        vector->doExport(source_vector, tpetra_export, Tpetra::REPLACE);
+        vector->doExport(source_vector, *tpetra_export, Tpetra::REPLACE);
       else if (operation == VectorOperation::add)
-        vector->doExport(source_vector, tpetra_export, Tpetra::ADD);
+        vector->doExport(source_vector, *tpetra_export, Tpetra::ADD);
       else
         AssertThrow(false, ExcNotImplemented());
     }
 
 
 
+    template <typename Number>
+    void
+    Vector<Number>::import_elements(
+      const ReadWriteVector<Number> &V,
+      VectorOperation::values        operation,
+      const std::shared_ptr<const Utilities::MPI::CommunicationPatternBase>
+        &communication_pattern)
+    {
+      const std::shared_ptr<const Utilities::MPI::CommunicationPatternBase>
+        local_comm_pattern =
+          std::make_shared<const Utilities::MPI::CommunicationPatternBase>(
+            *communication_pattern);
+
+      import_elements(V, operation, Teuchos::rcp(local_comm_pattern.get()));
```

Comment:
Why not just
```suggestion
      Utilities::MPI::CommunicationPatternBase>
        communication_pattern_copy = *communication_pattern);
      import_elements(V, operation, Teuchos::rcp(&communication_pattern_copy));
```

Reply:
```
Utilities::MPI::CommunicationPatternBase communication_pattern_copy = *communication_pattern;
```
That does not work. When I try it, I get the following errors:
```
</path/to/dealii>/dealii/include/deal.II/lac/trilinos_tpetra_vector.templates.h:298:9: error: cannot allocate an object of abstract type ‘dealii::Utilities::MPI::CommunicationPatternBase’
</path/to/dealii>/deal.II/dealii/include/deal.II/lac/trilinos_tpetra_vector.templates.h:298:9: error: cannot declare variable ‘communication_pattern_copy’ to be of abstract type ‘dealii::Utilities::MPI::CommunicationPatternBase’
```

Reply:
Hmm... The problem is that you are slicing the object with the current approach. Maybe, we should just deprecate the `std::shared_ptr` overload and ignore the `communication_pattern` argument since it's only considered an optimization anyway.

Reply:
That sounds good to me, and I marked the function as deprecated.